File: [Manya]/Leetcode/Sorting/1589.java
package Sorting;

import java.util.Arrays;

class Solution {
    public int maxSumRangeQuery(int[] nums, int[][] requests) {
        
        Arrays.sort(nums);

        int arr[] = new int[nums.length + 1];

        for(int i = 0; i < requests.length; i++){
            arr[requests[i][0]] += 1;
            arr[requests[i][1] + 1] -= 1;
        }
        for(int i = 1; i < arr.length; i++){
            arr[i] = arr[i] + arr[i-1];
        }
        Arrays.sort(arr);

        long sum = 0;

        for(int i = arr.length - 1; i > 0; i--){
            long a = nums[i-1];
            long b = arr[i];
            long ans = a * b;
            sum += ans;
        }
        return (int)(sum % 1000000007);
    }
}

File: Aditya-Pratap/Sorting/1589.py
class Solution:
    def maxSumRangeQuery(self, nums: List[int], requests: List[List[int]]) -> int:
        freq = [0] * (len(nums)+1)

        for start, end in requests:
            freq[start] += 1
            freq[end+1] -= 1
        
        prefix = [0] * len(nums)
        prefix[0] = freq[0]

        for i in range(1,len(nums)):
            prefix[i] = prefix[i-1] + freq[i]
        
        prefix.sort()
        nums.sort()
        summ = 0 

        for i in range(len(nums)):
            summ += (nums[i] * prefix[i]) % (10**9 + 7)

        return summ % (10**9 + 7)